site/content/contribute/developer-setup.md

```diff
@@ -0,0 +1,99 @@
+---
+title: "Developer setup"
+heading: "Set up your development environment"
+description: "Find out how to set up your development environment for building, running, and testing Mattermost."
+weight: 1
+aliases:
+  - /contribute/server/developer-setup
+  - /contribute/more-info/server/developer-setup/
+  - /contribute/more-info/webapp/developer-setup/
+---
+
+Set up your development environment for building, running, and testing Mattermost.
+
+{{<note "Warning" "icon-alert-outline">}}
+During the move to the monorepo, these instructions will be in flux. Follow https://github.com/mattermost/mattermost-server/issues/22420 for updates.
+{{</note>}}
+
+{{<note "Note:">}}
+If you're migrating from before the monorepo see the [migration notes]({{< ref "/contribute/monorepo-migration-notes" >}}).
+{{</note>}}
+
+{{<note "Note:">}}
+If you're developing plugins, see the plugin [developer setup]({{< ref "/integrate/plugins/developer-setup" >}}) documentation.
+{{</note>}}
+
+# Setup the Mattermost Server
+
+1. Install [Docker](https://www.docker.com/).
+
+1. Install [Go](https://go.dev/).
+
+1. Increase the number of available file descriptors. Update your shell's initialization script (e.g. `.bashrc` or `.zshrc`), and add the following:
+
+    ```sh
+    ulimit -n 8096
+    ```
+
+1. Fork https://github.com/mattermost/mattermost-server.
+
+1. Clone the Mattermost source code from your fork:
+
+    ```sh
+    git clone https://github.com/YOUR_GITHUB_USERNAME/mattermost-server.git
+    ```
+
+1. Start the server:
+
+    ```sh
+    cd server
+    make run-server
+    ```
+
+1. Test your environment:
+
+    ```sh
+    curl http://localhost:8065/api/v4/system/ping
+    ```
+
+    If successful, the `curl` step will return a JSON object:
+    ```json
+    {"AndroidLatestVersion":"","AndroidMinVersion":"","DesktopLatestVersion":"","DesktopMinVersion":"","IosLatestVersion":"","IosMinVersion":"","status":"OK"}
+    ```
+
+1. Stop the server:
+
+    ```sh
+    make stop-server
+    ```
+
+    The `stop-server` make target does not stop all the docker containers started by `run-server`. To stop the running docker containers:
+
+    ```sh
+    make stop-docker
+    ```
+
+1. Set your options:
+
+    Some behaviors can be customized such as running the server in the foreground as described in the `config.mk` file in the server directory. See that file for details.
+
+# Setup the Mattermost webapp
+
+1. Install NVM and use it to install the required version of Node.js:
```

Comment:
Perhaps we should move these steps to set up dependencies to be at the top as well? People aren't going to be installing or using just one or the other any more, so I think we may as well install them all in one go

Reply:
Yup. I was thinking something like this. I like the one shot instructions much better.